code/game/objects/items/weapons/storage/internal.dm

```diff
@@ -4,18 +4,22 @@
 	preserve_item = 1
 	var/atom/movable/master_atom
 
+
+/obj/item/storage/internal/Destroy()
+	master_atom = null
+	return ..()
+
+
 /obj/item/storage/internal/Initialize()
 	. = ..()
 	master_atom = loc
 	if(!istype(master_atom))
 		return INITIALIZE_HINT_QDEL
 	loc = master_atom
-	name = master_atom.name
+	if (name == "storage")
```

Comment:
I thought about this while writing it. The practical number of even stacked up world init time internal storage items is a couple handfuls of dozens- and those few uS are a one off. I will make this change, but I believe it will be sub-trivial if not detrimental.